$NetBSD: patch-ad,v 1.3 2009/11/15 03:11:12 snj Exp $

This is git revision 59eb52b7c6ccbe83742e5eb8b40b5e32bb6600cc upstream.

--- openbox/client.c.orig	2008-04-13 16:22:11.000000000 -0700
+++ openbox/client.c	2009-11-14 01:16:58.000000000 -0800
@@ -3897,12 +3897,15 @@ static void client_present(ObClient *sel
     client_focus(self);
 }
 
-/* this function exists to map to the client_activate message in the ewmh,
-   the user arg is unused because nobody uses it correctly anyway. */
+/* this function exists to map to the net_active_window message in the ewmh */
 void client_activate(ObClient *self, gboolean here, gboolean raise,
                      gboolean unshade, gboolean user)
 {
-    client_present(self, here, raise, unshade);
+    if (user || (self->desktop == DESKTOP_ALL ||
+                 self->desktop == screen_desktop))
+        client_present(self, here, raise, unshade);
+    else
+        client_hilite(self, TRUE);
 }
 
 static void client_bring_windows_recursive(ObClient *self,
